src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private static final double DEFAULT_SIZE_GROW = 1.5;
+
+    private int size;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
     @Override
     public void add(T value) {
-
+        checkSize();
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index != size) {
+            checkIndex(index);
+        }
+        checkSize();
+        T[] resultCopy = (T[]) new Object[values.length];
+        System.arraycopy(values, 0, resultCopy, 0, index + 1);
+        resultCopy[index] = value;
+        System.arraycopy(values, index, resultCopy, index + 1, size - index);
```

Comment:
same situation here, we should copy tail of array 1 position to the right and change value of index without creating new array or double usage of arraycopy() method